### ISSUE TITLE
Mounting Options

### ISSUE THREAD

**Issue by RobertD502** [jeroenterheerdt/neato-serial#5]
First of all thank you for creating this. I was able to utilize this to get my old Neato XV Signature Pro integrated into Home Assistant instead of having to go out and buy another Neato Botvac D5.

This post isn't about an issue per se- I was just curious as to how you went about mounting all of the devices to your neato. I noticed that if any of the devices (pi, relay, voltage regulator) are anywhere in the view of the LiDAR, the neato just goes in circles and fails to clean. So, I just wanted to see what kind of mounting you/anyone else has come up with to get around this. Mounting on top of the LiDAR within a box is an option, but that may be problematic as it will get knocked as the neato tries to go under tight spaces.

Thanks again for all the great work!

**Comment by jeroenterheerdt** [comment]
I mounted a RPi on the outside and everything else inside of the device. I think a RPi zero would even fit inside.

**Comment by jeroenterheerdt** [comment]
By the way, what did you do to make sure the USB connection is restored properly after switching the relay? I never got that to work properly with my xv, so interested in understanding your config

**Comment by RobertD502** [comment]
Just in case it matters- things I used:

1. Raspberry Pi Zero W
2. Relay: Youngneer 5v Relay Board Relay Module 

Amazon URL: https://www.amazon.com/dp/B07M88JRFY/ref=cm_sw_r_cp_api_i_NlgrFbERCEA1M

*Sorry for the formatting. On mobile and the link option isn’t working.

For the config file I edited the example file and kept all settings the same (then saved it without .example extension of course). I kept the setting to reboot the raspberry pi set to True. This has worked for me. I also kept the usb address(s) the same as in the example just in case the address decides to change to 1 instead of 0. 

Whenever I send a start command from HA, I can hear the relay click and the neato begins working. I can see the raspberry pi reboots at this point. If I manually cancel the cleaning process and carry the neato back to the charging base, I can see that the usb connection works WITHOUT me having to unplug the mini usb cable/reconnecting it and the neato properly reports that it is docked. I can issue a start command again without a problem- so as you can see the usb connection works fine without me physically intervening. 

Could it possibly be a problem with the relay that you are using? 

Only thing I am worried about is the possibility of wearing out the SD card with reboots every time a cleaning session is started. 

As a side note, I did notice that the mqtt service always threw errors and failed to start until I removed all comments from the config file.

**Comment by jeroenterheerdt** [comment]
Thanks and you toggled the power to the USB in the relay (the red cable)?

**Comment by RobertD502** [comment]
Yes. Red wire from micro USB side (since I am using a raspberry pi zero W) going into the COM port and red wire from mini usb side (neato side) going into NO port. The relay I am using is a 3 port relay.
